src/playermat/Playermat.ttslua

```diff
@@ -389,35 +389,29 @@ function doUpkeep(_, clickedByColor, isRightClick)
       local handCards = Player[playerColor].getHandObjects()
       local cardsToDiscard = {}
 
-      for i = 1, #handCards do
-        local md = JSON.decode(handCards[i].getGMNotes())
+      for _, card in ipairs(handCards) do
+        local md = JSON.decode(card.getGMNotes())
         if md ~= nil and (not md.weakness and not md.hidden and md.type ~= "Enemy") then
-          table.insert(cardsToDiscard, handCards[i])
+          table.insert(cardsToDiscard, card)
         end
       end
 
       -- perform discarding 1 by 1
       local pos = returnGlobalDiscardPosition()
-      local count = #cardsToDiscard
-      for i = count, 1, -1 do
-        Wait.time(function() deckLib.placeOrMergeIntoDeck(cardsToDiscard[i], pos, rot) end,
-          (count - i + 1) * 0.1)
-      end
+      deckLib.placeOrMergeIntoDeck(cardsToDiscard, pos, self.getRotation())
 
-      -- add some time if there are any cards to discard, if not, draw up to 5 immediately
-      local k = 0
-      if count > 0 then
-        k = 0.5 + (count * 0.1)
-      end
+      -- draw up to 5 cards
+      local cardsToDraw = 5 - #handCards + #cardsToDiscard
+      if cardsToDraw > 0 then
+        printToColor("Drawing " .. cardsToDraw .. " cards (Patrice)", messageColor)
```

Comment:
Message should probably mention discarding hand

Reply:
Updated